staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -419,3 +566,48 @@ func (p *processorListener) run(stopCh <-chan struct{}) {
 		}
 	}
 }
+
+// shouldResync deterimines if the listener needs a resync. If the listener's resyncPeriod is 0,
+// this always returns false.
+func (p *processorListener) shouldResync(now time.Time) bool {
+	if p.resyncPeriod == 0 {
+		return false
+	}
+
+	return now.After(p.nextResync) || now.Equal(p.nextResync)
+}
+
+// shouldHandleSyncItem returns true if the reflector has started a relisting, or if it has started
+// resyncing and the listener is participating in the current resync.
+func (p *processorListener) shouldHandleSyncItem() bool {
+	return p.relisting || p.resyncing && p.wantSyncItems
```

Comment:
It's because we need to know when we've popped off the last item from a resync, so that we can ignore future sync items until we hit the next resync period. However, as we discussed on irc, it sounds like making resync atomic would clean up this code significantly. I'll open up a separate pull for that (although it may be a few days before I can get to it).